src/GovUk.Education.ExploreEducationStatistics.Admin/Controllers/Api/UserManagement/UsersController.cs

```diff
@@ -28,77 +24,56 @@ public UsersController(IUserManagementService userManagementService)
 
         [HttpGet("user-management/users")]
         [ProducesResponseType(200)]
-        [ProducesResponseType(404)]
         public async Task<ActionResult<List<UserViewModel>>> GetUserList()
         {
-            var users = await _userManagementService.ListAsync();
-
-            if (users.Any())
-            {
-                return Ok(users);
-            }
-
-            return NotFound();
+            return await _userManagementService
+                .ListAllUsers()
+                .HandleFailuresOrOk();
         }
 
         [HttpGet("user-management/users/{userId}")]
         [ProducesResponseType(200)]
-        [ProducesResponseType(404)]
         public async Task<ActionResult<UserViewModel>> GetUser(string userId)
         {
-            var user = await _userManagementService.GetAsync(userId);
-
-            if (user != null)
-            {
-                return Ok(user);
-            }
-
-            return NotFound();
+            return await _userManagementService
+                .GetUser(userId)
+                .HandleFailuresOrOk();
         }
 
         [HttpPut("user-management/users/{userId}")]
-        public async Task<ActionResult<UserViewModel>> UpdateUser(string userId, EditUserViewModel model)
+        public async Task<ActionResult<Unit>> UpdateUser(string userId, EditUserViewModel model)
```

Comment:
`name` and `email` in UpdateUser model aren't doing anything which is a bit misleading.

I think ideally we would combine the UserRelease roles into the model as well and then scrap the additional endpoints `AddUserReleaseRole` and `DeleteUserReleaseRole` to follow a more RESTful design, coupling together the update of a user with their roles, after all this update is already allowing updating one type of Role.

I see here this it would be quite a big effort as there are additional things that hang off adding a role like sending an email, so maybe hold off on combining them for now.

Reply:
Created a separate ticket to address this: https://dfedigital.atlassian.net/browse/EES-1788